File: FlexTargetAndroid/app/src/main/java/com/flextarget/android/data/remote/ServerConfig.kt
package com.flextarget.android.data.remote

import android.content.Context
import android.content.SharedPreferences

class ServerConfig(private val context: Context) {

    private val prefs: SharedPreferences = context.getSharedPreferences("server_prefs", Context.MODE_PRIVATE)

    companion object {

        const val INTERNATIONAL = "https://app.etarget.grwolftactical.com"

        const val CHINA = "https://etarget.topoint-archery.cn"

    }

    fun getServerUrl(): String {

        return prefs.getString("server_url", INTERNATIONAL) ?: INTERNATIONAL

    }

    fun setServerUrl(url: String) {

        prefs.edit().putString("server_url", url).apply()

    }

    fun isInternational(): Boolean = getServerUrl() == INTERNATIONAL

    fun toggleServer() {

        val current = getServerUrl()

        setServerUrl(if (current == INTERNATIONAL) CHINA else INTERNATIONAL)

    }

}

File: FlexTargetiOS/flextarget/Services/ServerConfig.swift
import Foundation

class ServerConfig {

    private let userDefaults = UserDefaults.standard

    private let serverUrlKey = "serverUrl"

    static let international = "http://52.221.232.2"

    static let china = "http://124.222.233.30"

    func getServerUrl() -> String {

        return userDefaults.string(forKey: serverUrlKey) ?? ServerConfig.international

    }

    func initializeServer() {
        if userDefaults.string(forKey: serverUrlKey) == nil {
            let identifier = Locale.current.region?.identifier ?? ""
            if identifier == "CN" {
                setServerUrl(ServerConfig.china)
            } else {
                setServerUrl(ServerConfig.international)
            }
        }
    }

    func setServerUrl(_ url: String) {

        userDefaults.set(url, forKey: serverUrlKey)

    }

    func isInternational() -> Bool {

        return getServerUrl() == ServerConfig.international

    }

    func toggleServer() {

        let current = getServerUrl()

        setServerUrl(current == ServerConfig.international ? ServerConfig.china : ServerConfig.international)

    }

}